doc/ansible/dev/apb_migration_guide.md

```diff
@@ -0,0 +1,241 @@
+
+Next, add a finalizer block with a name of: `finalizer.<name>.<group>/<version>`, and set the playbook to point to your `deprovision.yml` in the container (likely location for that will be `/opt/ansible/playbooks/deprovision.yml`). For the memcached-operator we generated above, the watches.yaml would look like this:
+
+```yaml
+---
+- version: v1alpha1
+  group: apps.example.com
+  kind: Memcached
+  playbook: /opt/ansible/playbooks/provision.yml
+  finalizer:
+    name: finalizer.memcached.apps.example.com/v1alpha1
+    playbook: /opt/ansible/playbooks/deprovision.yml
+```
+
+##### Binding
+If you have a `bind` playbook, add a new entry to your `watches.yaml` (you can copy paste the existing entry). 
+
+The `version` and `group`, will remain unchanged, but update the `kind` with a `Binding` suffix. 
+
+For example, if you have a resource with `kind: Memcached`, the kind of your new entry will be `MemcachedBinding`. 
+
+The playbook for this entry should map to your `bind` playbook, (likely location `/opt/ansible/playbooks/bind.yml`), and if you have an `unbind` playbook then set the playbook for the finalizer to point to it (likely location `/opt/ansible/playbooks/unbind.yml`). If you don't have an `unbind` playbook, remove the finalizer block for your `Binding` resource.
+
+For an APB with both `bind` and `unbind` playbooks, the `watches.yaml` would end up looking like this:
+
+```yaml
+---
+- version: v1alpha1
+  group: apps.example.com
+  kind: Memcached
+  playbook: /opt/ansible/playbooks/provision.yml
+  finalizer:
+    name: finalizer.memcached.apps.example.com/v1alpha1
+    playbook: /opt/ansible/playbooks/deprovision.yml
+- version: v1alpha1
+  group: apps.example.com
+  kind: MemcachedBinding
+  playbook: /opt/ansible/playbooks/bind.yml
+  finalizer:
+    name: finalizer.memcachedbinding.apps.example.com/v1alpha1
+    playbook: /opt/ansible/playbooks/unbind.yml
+```
+
+
+You will also need to run `operator-sdk add crd --api-version=<group>/<version> --kind=<kind>` to generate a new CRD and example in `deploy/crds`.
+
+#### deploy/crds/
+Now that you have all your CRDs created, you can generate the OpenAPI spec for them using your `apb.yml`. 
+
+The `convert.py` script included at the bottom of this document can handle the conversion to the OpenAPI spec, at which point you can copy paste everything from `validation:` on into your primary CRD (for the regular `parameters`), or into your `Binding` CRD (for `bind_parameters`).
+
+You may notice that the OpenAPI validation uses `camelCase` parameters, while your `apb.yml` and Ansible playbooks probably assume `snake_case` variables. `Ansible Operator` will automatically convert the `camelCase` parameters from the Kubernetes resource into `snake_case` before passing them to your playbook, so this should not require any change on your part.
```

Comment:
What happens if they run convert.py? what should the output look like? What should they expect to see after it runs? We should expand on this section.